### ISSUE TITLE
PySide 6.8 compatibility concerns?

### ISSUE THREAD

**Issue by hmaarrfk** [jupyter/qtconsole#624]
@ccordoba12 i know this is your favorite set of issues but here goes.

qtconsole on the master branch
```
  File "/home/mark/git/qtconsole/qtconsole/bracket_matcher.py", line 90, in _cursor_position_changed
    self._text_edit.setExtraSelections([])
AttributeError: type object 'PySide6.QtWidgets.QTextEdit' has no attribute 'ExtraSelection'. Did you mean: 'extraSelections'
```

But I first wanted to report upstream, but when I tried the following
```python
import sys
from qtpy.QtWidgets import QApplication, QTextEdit
from qtpy.QtCore import QTimer
# from PySide6.QtWidgets import QApplication, QTextEdit
# from PySide6.QtCore import QTimer

def update_text(text_edit):
    text_edit.setPlainText("hello world")

app = QApplication(sys.argv)

text_edit = QTextEdit()
text_edit.setPlainText("hello")
text_edit.show()

timer = QTimer()

def hello():
    print("Hello")
    text_edit.setExtraSelections([])
    print("done")

timer.singleShot(500, hello)

sys.exit(app.exec())
```

and couldn't reproduce the error.

I'll try to narrow it down further but this is as far as I got now.

**Comment by hmaarrfk** [comment]
And actually, from pypi it just segfaults:

```
conda create --name test python=3.12
conda activate test
# Install all deps
pip install -e . -vv
# Will install pyside 6.8.0.1
pip install PySide6
```

```
# 6.7.0 also Just segfaults
pip install PySide6==6.7.0

# 6.7.1 seems to works
pip install PySide6==6.7.1
```

**Comment by hmaarrfk** [comment]
The following patch seems to "resolve" things:
```patch
diff --git a/qtconsole/bracket_matcher.py b/qtconsole/bracket_matcher.py
index fd593f7..82eb70f 100644
--- a/qtconsole/bracket_matcher.py
+++ b/qtconsole/bracket_matcher.py
@@ -87,7 +87,7 @@ class BracketMatcher(QtCore.QObject):
         """ Updates the document formatting based on the new cursor position.
         """
         # Clear out the old formatting.
-        self._text_edit.setExtraSelections([])
+        self._text_edit.extraSelections = []
 
         # Attempt to match a bracket for the new cursor position.
         cursor = self._text_edit.textCursor()
@@ -97,4 +97,4 @@ class BracketMatcher(QtCore.QObject):
             if match_position != -1:
                 extra_selections = [ self._selection_for_character(pos)
                                      for pos in (position, match_position) ]
-                self._text_edit.setExtraSelections(extra_selections)
+                self._text_edit.extraSelections = extra_selections
diff --git a/qtconsole/console_widget.py b/qtconsole/console_widget.py
index aa492ba..e8aa4fb 100644
--- a/qtconsole/console_widget.py
+++ b/qtconsole/console_widget.py
@@ -1201,6 +1201,7 @@ class ConsoleWidget(MetaQObjectHasTraits('NewBase', (LoggingConfigurable, superQ
             control = QtWidgets.QPlainTextEdit()
         elif self.kind == 'rich':
             control = QtWidgets.QTextEdit()
+            control.extraSelections = []
             control.setAcceptRichText(False)
             control.setMouseTracking(True)

```

But I just can't really understand why the following works:
```
>>> from qtpy.QtWidgets import QApplication, QTextEdit
A known critical bug in PySide6 6.8.0 will cause your application to crash see https://github.com/spyder-ide/qtpy/issues/494
a
>>> app = QApplication()
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
>>> text_edit = QTextEdit()
>>> text_edit.setExtraSelections([])
>>> import qtpy
>>> qtpy.API
'pyside6'
```

I can't help to think that something else is going on???

**Comment by hmaarrfk** [comment]
Things seem to be resolved in PySide6 6.8.0.2
